File: sistem/ORM/atendimento.py
from sistem.ORM.model.models import Atendimento, set_db
import eel

@eel.expose
def ate_create(ate_descricao:str, ate_categoria:int, ate_cliente:int, ate_status:int, ate_data_criacao:str|None = None)->str:
    db = set_db()
    with db.connection():
        try:
            atendimento = Atendimento(ate_data_criacao=ate_data_criacao, ate_descricao=ate_descricao, ate_categoria=ate_categoria, ate_cliente=ate_cliente, ate_status=ate_status)
            atendimento.save()
            return "Atendimento criado com sucesso!"
        except Exception as e:
            print(e)
            return "erro ao criar o atendimento"
    
@eel.expose
def ate_read_all():
    db = set_db()
    with db.connection():
        try:
            ate = []
            for atendimento in Atendimento.select():
                print(atendimento.ate_descricao)
                ate.append(atendimento.ate_descricao)
            return ate
        except Exception as e:
            print(e)
            return "Erro ao ler os atendimentos"
    
@eel.expose
def ate_update(ate_id:int, ate_descricao:str, ate_categoria:int, ate_cliente:int, ate_status:int, ate_data_criacao:str|None = None)->str:
    db = set_db()
    with db.connection():
        try:
            atendimento = Atendimento.get(Atendimento.id == ate_id)
            atendimento.ate_data_criacao = ate_data_criacao
            atendimento.ate_descricao = ate_descricao
            atendimento.ate_categoria = ate_categoria
            atendimento.ate_cliente = ate_cliente
            atendimento.ate_status = ate_status
            atendimento.save()
            return "Atendimento atualizado com sucesso!"
        except Exception as e:
            print(e)
            return "Erro ao atualizar o atendimento"
    
@eel.expose
def ate_delete(ate_id:int)->str:
    db = set_db()
    with db.connection():
        try:
            atendimento = Atendimento.get(Atendimento.id == ate_id)
            atendimento.delete_instance()
            return "Atendimento deletado com sucesso!"
        except Exception as e:
            print(e)
            return "Erro ao deletar o atendimento"

@eel.expose
def ate_search(ate_descricao:str)->str:
    db = set_db()
    with db.connection():
        try:
            atendimento = Atendimento.select().where(Atendimento.ate_descricao.contains(ate_descricao))
            return atendimento
        except Exception as e:
            print(e)
            return "Erro ao pesquisar o atendimento"
        
eel.init('web')

File: web/js/atendimento.js
async function cadastrarAtendimento() {
    const ate_data = document.getElementById("ate_data").value;
    const ate_cliente = document.getElementById("ate_cliente").value;
    const ate_descricao = document.getElementById("ate_descricao").value;
    const ate_categoria = document.getElementById("ate_categoria").value;
    const ate_status = document.getElementById("ate_status").value;

    if (ate_data == "") {
        const resultado = await eel.ate_create(ate_descricao, ate_categoria, ate_cliente, ate_status)();
    }

    const resultado = await eel.ate_create(ate_data, ate_hora, ate_cliente, ate_descricao)();

    alert(resultado);
}

async function listarAtendimentos() {
    const resultado = await eel.ate_read_all()();
    console.log(resultado);
}

async function atualizarAtendimento() {
    const ate_id = document.getElementById("ate_id").value;
    const ate_data = document.getElementById("ate_data").value;
    const ate_cliente = document.getElementById("ate_cliente").value;
    const ate_descricao = document.getElementById("ate_descricao").value;
    const ate_categoria = document.getElementById("ate_categoria").value;
    const ate_status = document.getElementById("ate_status").value;

    const resultado = await eel.ate_update(ate_id, ate_data, ate_cliente, ate_descricao, ate_categoria, ate_status)();

    alert(resultado);
}

async function deletarAtendimento() {
    const ate_id = document.getElementById("ate_id").value;

    const resultado = await eel.ate_delete(ate_id)();

    alert(resultado);
}

async function buscarAtendimento() {
    const ate_id = document.getElementById("ate_id").value;

    const resultado = await eel.ate_read(ate_id)();

    alert(resultado);
}
